src/platform/forms-system/src/js/components/ProgressButton.jsx

```diff
@@ -85,6 +85,9 @@ ProgressButton.propTypes = {
   // button
   ariaLabel: PropTypes.string,
 
+  // aria-describedby attribute; used to associate the button with another element
+  ariaDescribedBy: PropTypes.string,
```

Comment:
We are modifying this file which is already very similar to the `ProgressButton` that is in the component library.

<details>

<summary>Diff of both ProgressButton files</summary>

```diff
--- src/platform/forms-system/src/js/components/ProgressButton.jsx	2022-05-27 14:28:10.310423053 -0700
+++ ../component-library/packages/react-components/src/components/ProgressButton/ProgressButton.jsx	2022-04-05 14:29:08.068709160 -0700
@@ -1,18 +1,19 @@
 import PropTypes from 'prop-types';
 import React from 'react';
-import uniqueId from 'lodash/uniqueId';
+import { uniqueId } from '../../helpers/utilities';
+import classNames from 'classnames';
 
 /**
  * A component for the continue button to navigate through panels of questions.
  */
 
 class ProgressButton extends React.Component {
-  /* eslint-disable-next-line camelcase */
   UNSAFE_componentWillMount() {
     this.id = uniqueId();
   }
 
   render() {
+    const { ariaLabel = null, ariaDescribedby = null, disabled } = this.props;
     const beforeText = this.props.beforeText ? (
       <span className="button-icon" aria-hidden="true">
         {this.props.beforeText}
@@ -31,19 +32,16 @@
     );
 
     return (
-      // aria-describedby tag to match "By submitting this form"
-      // text for proper screen reader operation
       <button
         type={this.props.submitButton ? 'submit' : 'button'}
-        disabled={this.props.disabled}
-        className={`${this.props.buttonClass}${
-          this.props.disabled ? ' usa-button-disabled' : ''
-        }`}
+        disabled={disabled}
+        className={classNames(this.props.buttonClass, {
+          'usa-button-disabled': disabled,
+        })}
         id={`${this.id}-continueButton`}
         onClick={this.props.onButtonClick}
-        onMouseDown={this.props.preventOnBlur}
-        aria-label={this.props.ariaLabel || null}
-        aria-describedby={this.props.ariaDescribedBy}
+        aria-label={ariaLabel}
+        aria-describedby={ariaDescribedby}
       >
         {beforeText}
         {this.props.buttonText}
@@ -53,39 +51,50 @@
   }
 }
 
-ProgressButton.defaultProps = {
-  preventOnBlur: e => {
-    e.preventDefault();
-  },
-};
-
 ProgressButton.propTypes = {
-  // function that changes the path to the next panel or submit.
+  /**
+   * Function called when the button is clicked.
+   */
   onButtonClick: PropTypes.func,
 
-  // function that bypasses onBlur when clicking
-  preventOnBlur: PropTypes.func,
-
-  // what is the button's label
+  /**
+   * The text on the button
+   */
   buttonText: PropTypes.string.isRequired,
 
-  // what CSS class(es) does the button have
-  buttonClass: PropTypes.string.isRequired,
-
-  // Stores the value for the icon that will appear before the button text.
+  /**
+   * CSS class(es) to apply to the button
+   */
+  buttonClass: PropTypes.string,
+
+  /**
+   * A character to appear before the button text. This is wrapped in a `span.button-icon`.
+   */
   beforeText: PropTypes.string,
 
-  // Stores the value for the icon that will appear after the button text.
+  /**
+   * A character to appear after the button text. This is wrapped in a `span.button-icon`.
+   */
   afterText: PropTypes.string,
 
-  // is the button disabled or not
+  /**
+   * Whether the button is disabled
+   */
   disabled: PropTypes.bool,
 
-  // aria-label attribute; needed for the review & submit page "Update page"
-  // button
+  /**
+   * Text read by a screenreader instead of content within the button
+   */
   ariaLabel: PropTypes.string,
 
-  // is this a submit button or not
+  /*
+   * Element ID containing additional content read by a screenreader
+   */
+  ariaDescribedby: PropTypes.string,
+
+  /**
+   * Whether the button is a submit button
+   */
   submitButton: PropTypes.bool,
 };
```

</details>
 
 Unless there is a good reason to keep them separate, I think we should remove this file, use the component from the component-library, and eliminate any further opportunity for version skew.
 
 Things that come to mind as potential sticking points with that based on the diff:
 
 1. The different `uniqueId` function might no longer give a `uniqueId`, but I think the Cypress axe tests would let us know about that.
 2. The component-library `ProgressButton` doesn't prevent the `mousedown` event from propagating by default

Reply:
I would be fine without the changes to the ProgressButton files because they basically amount to code duplication.

Reply:
This pull request adds a missing propType and updates a unit test for the ProgressButton component for `aria-describedby`, which has been in there since [this pull request](https://github.com/department-of-veterans-affairs/vets-website/pull/19326/files#diff-d51faea197ba71f8b7a2f2f39d4b642cc598bd911e1e62ea00abf3b87b387850R46) from last November. That isn't code duplication. Additionally, the sticking points mentioned above make it seem like it wouldn't be just a 1-1 drop-in and replace.

Reply:
I called it code duplication because it duplicates code that exists in the component library:

- [the propType](https://github.com/department-of-veterans-affairs/component-library/blob/13b80bda04777123ef7a0420f01ac7d69804356b/packages/react-components/src/components/ProgressButton/ProgressButton.jsx#L90-L93)
- [the unit test](https://github.com/department-of-veterans-affairs/component-library/blob/13b80bda04777123ef7a0420f01ac7d69804356b/packages/react-components/src/components/ProgressButton/ProgressButton.unit.spec.jsx#L26)

I understand that those are both minor changes. However, I still think it's best to freeze both components or to remove one of them.

Reply:
I've created [this ticket](https://app.zenhub.com/workspaces/forms-library---platform-spike-team-61b0ae1f2cd3c30014e8a5b0/issues/department-of-veterans-affairs/va-forms-system-core/427) for the work to replace the above component with the one from the component library. Like I mentioned earlier, I agree with you that it should be replaced, it just shouldn't be folded into this pull request. 

I'm not sure when the work for replacing the component will get prioritized, or which team should be responsible for it, so for now the propType and the unit test updates should stay, as they align with good practice; props should have propTypes, and changes should be tested.

Reply:
Thanks for creating that ticket @micahchiang. 

@bkjohnson, I have reverted my changes to ProgressButton and have just left the changes to add the two aria-describedBy attributes. These changes should be sufficient to close out my ticket. Thank you!